webapp/channels/src/components/admin_console/access_control/editors/table_editor/value_selector_menu.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import type {PropertyFieldOption} from '@mattermost/types/properties';
+
+import MultiValueSelector from './multi_value_selector_menu';
+import SingleValueSelector from './single_value_selector_menu';
+
+export interface TableRow {
+    attribute: string;
+    operator: string;
+    values: string[];
+}
+
+export interface ValueSelectorMenuProps {
+    row: TableRow;
+    disabled: boolean;
+    updateValues: (values: string[]) => void;
+    options?: PropertyFieldOption[];
+    allowCreateValue?: boolean;
+    placeholder?: string;
+}
+
+// Main ValueSelectorMenu component that delegates to the appropriate selector
+const ValueSelectorMenu = ({
```

Comment:
> This component should be split into two different components, one for single-value mode and another for multi-value. Right now, it seems like these first 4 lines are the only things shared between them, and the rest is mixed between the two modes

https://github.com/mattermost/mattermost/pull/30785#discussion_r2088992022 comment from @hmhealey 

I did split into two components, it was cleaner to have some sort of component to handle boilerplate code here.